plugins/field-multilineinput/src/field_multilineinput.ts

```diff
@@ -0,0 +1,521 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Multiline text input field.
+ */
+
+import * as Blockly from 'blockly/core';
+
+
+/**
+ * Class for an editable text area input field.
+ */
+export class FieldMultilineInput extends Blockly.FieldTextInput {
+  /**
+   * The SVG group element that will contain a text element for each text row
+   *     when initialized.
+   */
+  textGroup: SVGGElement|null = null;
+
+  /**
+   * Defines the maximum number of lines of field.
+   * If exceeded, scrolling functionality is enabled.
+   */
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  protected maxLines_ = Infinity;
+
+  /** Whether Y overflow is currently occurring. */
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  protected isOverflowedY_ = false;
+
+  /**
+   * @param value The initial content of the field.  Should cast to a string.
+   *     Defaults to an empty string if null or undefined.  Also accepts
+   *     Field.SKIP_SETUP if you wish to skip setup (only used by subclasses
+   *     that want to handle configuration and setting the field value after
+   *     their own constructors have run).
+   * @param validator An optional function that is called to validate any
+   *     constraints on what the user entered.  Takes the new text as an
+   *     argument and returns either the accepted text, a replacement text, or
+   *     null to abort the change.
+   * @param config A map of options used to configure the field.
+   *     See the [field creation documentation]{@link
+   * https://developers.google.com/blockly/guides/create-custom-blocks/fields/built-in-fields/multiline-text-input#creation}
+   * for a list of properties this parameter supports.
+   */
+  constructor(
+      value?: string|typeof Blockly.Field.SKIP_SETUP,
+      validator?: FieldMultilineInputValidator,
+      config?: FieldMultilineInputConfig) {
+    super(Blockly.Field.SKIP_SETUP);
+
+    if (value === Blockly.Field.SKIP_SETUP) return;
+    if (config) {
+      this.configure_(config);
+    }
+    this.setValue(value);
+    if (validator) {
+      this.setValidator(validator);
+    }
+  }
+
+  /**
+   * Configure the field based on the given map of options.
+   *
+   * @param config A map of options to configure the field based on.
+   */
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  protected override configure_(config: FieldMultilineInputConfig) {
+    super.configure_(config);
+    if (config.maxLines) this.setMaxLines(config.maxLines);
+  }
+
+  /**
+   * Serializes this field's value to XML.
+   * Should only be called by Blockly.Xml.
+   *
+   * @param fieldElement The element to populate with info about the field's
+   *     state.
+   * @returns The element containing info about the field's state.
+   * @internal
```

Comment:
Remove `@internal` tag, here and below, since these will be called by code outside of this plugin.